spec/i18n/placeholders_spec.rb

```diff
@@ -0,0 +1,45 @@
+describe "Placeholders in strings" do
+  it "translations preserve placeholders in strings" do
+    dirs = [Rails.root.join('locale')]
+
+    Rails::Engine.subclasses.each do |engine|
+      dir = engine.root.join('locale')
+      dirs << dir if dir.exist?
+    end
+
+    errors = {}
```

Comment:
Prefer `errors = Hash.new { |h, k| h[k] = [] }`.  This will allow you to avoid line 25 below with the `||=`